File: core/extract/repository_extractor.py
from __future__ import annotations

import re


def extract_repository_data(text: str, source_url: str = "") -> dict:
    src = text or ""
    tree_lines = [ln.strip() for ln in src.splitlines() if "/" in ln or "." in ln]
    readme_sections = sorted(set(re.findall(r"^#{1,6}\s+(.+)$", src, flags=re.MULTILINE)))
    return {
        "repository": {"source": source_url, "structure_hints": sorted(set(tree_lines))[:200]},
        "readme": {"sections": readme_sections},
    }

File: src/extract/repositoryExtractor.ts
/**
 * Converted from Python: core/extract/repository_extractor.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function extractRepositoryData(text: any, source_url: any = ""): any {
  var src: any = py.or2(text, () => (""));
  var tree_lines: any = py.iter(py.splitlines(src)).filter((ln: any) => (py.contains(ln, "/") || py.contains(ln, "."))).map((ln: any) => py.strip(ln));
  var readme_sections: any = py.sorted(py.toSet(py.reFindall("^#{1,6}\\s+(.+)$", src, "m")));
  return {"repository": {"source": source_url, "structure_hints": py.slice(py.sorted(py.toSet(tree_lines)), null, 200)}, "readme": {"sections": readme_sections}};
}
